sparsecoding/inference.py

```diff
@@ -269,4 +269,112 @@ def infer(self, data, dictionary):
             
             self.checknan(a,'coefficients')
         return a
+    
+    
+    
+class LSM(InferenceMethod):
+    """
+    Infer coefficients for each image in data using elements dictionary.
+    Method implemented according to "Group Sparse Coding with a Laplacian Scale Mixture Prior" (P. J. Garrigues & B. A. Olshausen, 2010)
+    """    
+
+    def __init__(self, n_iter = 100, n_iter_LSM = 6, BETA=0.01, ALPHA=80.0, sigma=0.005, sparseThreshold = 10**-2, solver = None):
+        '''
+        
+        Parameters
+        ----------
+        n_iter : scalar (1,) default=100
+            number of iterations to run for an optimizer
+
+        n_iter_LSM : scalar (1,) default=6
+            number of iterations to run the outer loop of  LSM
+
+        BETA : scalar (1,) default=0.01
+            LSM parameter used to update LAMBDA          
+
+        ALPHA : scalar (1,) default=80.0
+            LSM parameter used to update LAMBDA  
+
+        sigma : scalar (1,) default=0.005
+            LSM parameter used to compute the loss function
+
+        sparseThreshold : scalar (1,) default=10**-2
+            threshold used to discard smallest coefficients in the final solution
+            SM parameter used to compute the loss function
+            
+        solver : default=None
+        '''
+        super().__init__(solver)
+        self.n_iter = n_iter
+        self.n_iter_LSM = n_iter_LSM
+        
+        self.BETA = BETA
+        self.ALPHA = ALPHA
+        self.sigma = sigma
+        self.sparseThreshold = sparseThreshold
+
+    def LSMLoss(self, Phi, x, s, Lambda, sigma):
```

Comment:
All functions should have names which are lower case / snake case.

LSMLoss -> lsm_loss